test/k8sT/Services.go

```diff
@@ -1078,6 +1146,44 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 					SkipItIf(helpers.DoesNotExistNodeWithoutCilium, "Tests externalIPs", func() {
 						testExternalIPs(kubectl, ni)
 					})
+					SkipContextIf(helpers.DoesNotRunOn54OrLaterKernel, "Test NodePort with netfilterCompatMode=true", func() {
```

Comment:
> Regarding, using a single helper function for moving the test spec is bit more trouble. I had tried using a helper function to move code in this spec into a helper function, somehow infra doesn't like it. At runtime, when i pass the kubectl var as an argument to helper function, the helper function get always nil value for kubectl. Rather i prefer leave the test code as it is, current form is much more readable.

I disagree. It's possible to refactor this code to avoid duplicating it three times. If your refactoring causes issues, then we need to debug those issues. I'm happy to help on Slack of course!

In particular, there are two choices here I think:
1. Define a single helper function for the whole block passed to `SkipContextIf`. That function will have to take the Cilium configuration in argument and maybe a flag for the `testNodePort` statement (depends on why those differ between the different `SkipContextIf` blocks).
2. Define two helper functions, for the `BeforeAll` and `AfterAll` blocks.

Reply:
Refactored test code, added a  new function `installCiliumAndWaitForToComplete` in `test/k8sT/service_helpers.go` to move some code in Services.go